### ISSUE TITLE
Packed (interlaced) audio data

### ISSUE THREAD

**Issue by lp35** [x42/libltc#32]
Hi,

Currently using Portaudio for retrieving audio signal, I saw that ltc decoder does not support interlaced data (for 16bits for instance). I had to create my own template.

I have made a working prototype for s16i template. Any chance to see this update upstream?

**Comment by x42** [comment]
Interleaved (not interlaced)?

With Portaudio the easiest is to just use the `paNonInterleaved` PaSampleFormat. Portaudio can also directly provide `paUInt8`.

I'm not opposed to adding adding more formats, but I don't really see the point to add all possible combinations. Support for float and 16bit was added "make the common case fast". 

I have no interested to write code for other cases, it is easy to convert/de-interleave at application level and call  `ltc_decoder_write()`,  but you're very welcome to file a pull-request.

I suppose adding a `stride` parameter to the `LTCWRITE_TEMPLATE` define can work and simply skip over samples (and the application provides a buffer that is already offset to the given channel).

How did you implement it?

**Comment by lp35** [comment]
Yes I was meaning interlaced, paNonInterleaved in portaudio misled me. Thanks!

1. Concerning your portaudio suggestion, my pipeline need to use interlaced data, I can't add paNonInterleaved.

2. I agree with you regarding not supporting all the format the world can carry. However, depending on the rollout of https://github.com/x42/libltc/issues/31, I think it could be a good thing to move the template LTCWRITE_TEMPLATE from ltc.c to ltc.h, so the users can implement their own conversion without recompiling ltc.

3. Agree on stride parameter.


4. Here you can see my implementation for short data interlaced:
https://github.com/lp35/libltc/blob/master/src/ltc.c#L97
Note: I really like how you did implement the template, on a coding style perspective.

**Comment by lp35** [comment]
Hi,

I re-up this question, I am ready to invest some time for this. Any thought on my last comments?

Thank youm BR
